app/services/imports/lettings_logs_field_import_service.rb

```diff
@@ -203,5 +205,103 @@ def update_reason(xml_doc)
+        return @logger.info("lettings log #{record.id} has all household member details, skipping update")
+      end
+      if record.hhmemb == 8 || (record.hhmemb + 1..8).none? { |person_index| file_contains_person_details?(xml_doc, person_index) }
+        return @logger.info("lettings log #{record.id} has no additional household member details, skipping update")
+      end
+
+      person_index = 2
+      next_person_index = person_index + 1
+
+      while person_index <= record.hhmemb
+        if next_person_index <= record.hhmemb
+          if record.has_any_person_details?(person_index) || record["details_known_#{person_index}"] == 1
+            @logger.info("lettings log #{record.id} has details for person #{person_index}, skipping update")
+            person_index += 1
+            next_person_index += 1
+            next
+          end
+
+          if !record.has_any_person_details?(next_person_index) && record["details_known_#{next_person_index}"] != 1
+            next_person_index += 1
+            next
+          end
+
+          record["details_known_#{person_index}"] = record["details_known_#{next_person_index}"]
+          record["age#{person_index}"] = record["age#{next_person_index}"]
+          record["age#{person_index}_known"] = record["age#{next_person_index}_known"]
+          record["sex#{person_index}"] = record["sex#{next_person_index}"]
+          record["ecstat#{person_index}"] = record["ecstat#{next_person_index}"]
+          record["relat#{person_index}"] = record["relat#{next_person_index}"]
+
+          record["details_known_#{next_person_index}"] = nil
+          record["age#{next_person_index}"] = nil
+          record["age#{next_person_index}_known"] = nil
+          record["sex#{next_person_index}"] = nil
+          record["ecstat#{next_person_index}"] = nil
+          record["relat#{next_person_index}"] = nil
+
+          @logger.info("lettings log #{record.id}'s person #{next_person_index} details moved to person #{person_index} details")
+
+          person_index += 1
+          next_person_index += 1
+        else
+          record["age#{person_index}"] = safe_string_as_integer(xml_doc, "P#{next_person_index}Age")
+          record["age#{person_index}_known"] = age_known(xml_doc, next_person_index)
+          record["sex#{person_index}"] = sex(xml_doc, next_person_index)
+          record["ecstat#{person_index}"] = unsafe_string_as_integer(xml_doc, "P#{next_person_index}Eco")
+          record["relat#{person_index}"] = relat(xml_doc, next_person_index)
+          record["details_known_#{person_index}"] = details_known(person_index, record)
+          @logger.info("lettings log #{record.id}, reimported person #{person_index} details")
+          person_index += 1
+          next_person_index += 1
+        end
+      end
+
+      record.values_updated_at = Time.zone.now
```

Comment:
I'd only set this if values have changed (unless we can guarantee that at this point but I'm not sure we can)

Reply:
Hmm, I think we can? Cause we only get it to hear if there are additional details for household members and there is at least 1 person missing details, which means the people would shuffle. LMK if I'm missing a case tho

Reply:
Yep you're right :D